multitenancy-spring-boot-autoconfigure/src/test/java/org/axonframework/extensions/multitenancy/autoconfig/MultiTenantDataSourceManagerTest.java

```diff
@@ -66,48 +62,75 @@ class MultiTenantDataSourceManagerTest {
-                          .withBean("tenantDataSourceResolver", Function.class, () -> tenantDataSourceResolver)
-                          .withBean("properties", DataSourceProperties.class, () -> defaultDataSourceProperties)
-                          .run(context -> assertThat(context)
-                                  .getBean("multiTenantDataSourceManager")
-                                  .returns(MultiTenantDataSourceManager.class, bean -> {
-                                      MultiTenantDataSourceManager multiTenantDataSourceManager = ((MultiTenantDataSourceManager) bean);
-
-                                      verify(tenantProvider).subscribe(multiTenantDataSourceManager);
-
-                                      DriverManagerDataSource actualDataSource =
-                                              (DriverManagerDataSource) ((MultiTenantDataSourceManager) bean).getMultiTenantDataSource()
-                                                                                                             .getResolvedDefaultDataSource();
-                                      //noinspection ConstantConditions
-                                      assertEquals(expectedDataSource.getUrl(), actualDataSource.getUrl());
-                                      assertEquals(expectedDataSource.getUsername(), actualDataSource.getUsername());
-                                      assertEquals(expectedDataSource.getPassword(), actualDataSource.getPassword());
-
-                                      return MultiTenantDataSourceManager.class;
-                                  }));
+                .withAllowBeanDefinitionOverriding(true)
+                .withBean(TenantProvider.class, () -> tenantProvider)
+                .withBean("tenantDataSourceResolver", Function.class, () -> tenantDataSourceResolver)
+                .withBean("properties", DataSourceProperties.class, () -> defaultDataSourceProperties)
+                .run(context -> {
+                    assertThat(context).hasSingleBean(MultiTenantDataSourceManager.class);
+                    MultiTenantDataSourceManager multiTenantDataSourceManager = context.getBean(MultiTenantDataSourceManager.class);
+
+                    verify(tenantProvider).subscribe(multiTenantDataSourceManager);
+                    assertThat(TestConfiguration.dataSourceResolved.get());
+                });
     }
+
+    @Test
+    void resolveTenantDataSource() {
+        DataSourceProperties defaultDataSourceProperties = mock(DataSourceProperties.class);
+
+        TenantProvider tenantProvider = mock(TenantProvider.class);
+        when(tenantProvider.subscribe(any())).thenReturn(() -> true);
+
+        this.contextRunner
+                .withPropertyValues("axon.axonserver.contexts=default")
+                .withAllowBeanDefinitionOverriding(true)
+               // .withUserConfiguration(DataSourceResolverConfiguration.class)
```

Comment:
Is the `withUserConfiguration(...)` not used at all? So, the `@Configuraiton static`  class' are automatically added like this?